modules/relay/src/main/RelayNotifier.scala

```diff
@@ -0,0 +1,19 @@
+package lila.relay
+
+final private class RelayNotifier(notifyApi: lila.notify.NotifyApi, tourRepo: RelayTourRepo)(using Executor):
+
+  def roundBegin(rt: RelayRound.WithTour): Funit =
+    tourRepo.hasNotified(rt) collect:
```

Comment:
`collect` is not right here. It fails the future if there's no match.

You "solved" this by ignoring the future result completely at its invocation point in `createOrUpdateChapter` but that's not the way to go. Future should always be chained, not ignored.

Reply:
is this something that you mean just generally or is there a functional reason why ignoring a failure of hasNotified matters?

Reply:
It's both. In general it's important to chain futures, and here specifically it makes sense for the `RelayNotifier` to return a failed future if it can't do `hasNotified`, instead of sweeping a potential error under the carpet.